docs/site/src/pages/style/color/Color.js

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import { createStyleSheet } from 'jss-theme-reactor';
+import * as colors from 'material-ui/styles/colors';
+import customPropTypes from 'material-ui/utils/customPropTypes';
+import { getContrastRatio } from 'material-ui/styles/colorManipulator';
+
+const mainColors = [
+  'Red', 'Pink', 'Purple', 'Deep Purple', 'Indigo', 'Blue', 'Light Blue',
+  'Cyan', 'Teal', 'Green', 'Light Green', 'Lime', 'Yellow', 'Amber', 'Orange',
+  'Deep Orange',
+];
+const neutralColors = ['Brown', 'Grey'];
+
+export const styleSheet = createStyleSheet('colors', (theme) => ({
+  name: {
+    display: 'block',
+    marginBottom: 60,
+  },
+  colorContainer: {
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+  blockSpace: {
+    height: '4px',
+    color: '#eeeeee',
+  },
+  colorValue: {
+    fontSize: 12,
+  },
+  root: {
+    marginLeft: '50px',
+    boxSizing: 'initial',
+  },
+  hex: {
+    float: 'right',
+  },
+  colorGroup: {
+    float: 'left',
+    padding: '16px 0',
+    display: 'block',
+    margin: 0,
+    width: '30% ',
+    borderRight: 'solid 15px',
+    borderColor: '#eeeeee',
+  },
+  headline: {
+    fontSize: 24,
+    lineHeight: '32px',
+    paddingTop: 16,
+    marginBottom: 12,
+    letterSpacing: 0,
+    fontWeight: theme.typography.headline.fontWeight,
+    color: theme.typography.headline.color,
+  },
+}));
+
+export default function Color(props, context) {
+  const classes = context.styleManager.render(styleSheet);
+  let colorGroups = [];
+  let neutralGroups = [];
+
+  function getColorGroup(styles, color, showAltPalette) {
+    const mainPalette = [50, 100, 200, 300, 400, 500, 600, 700, 800, 900];
+    const altPalette = ['A100', 'A200', 'A400', 'A700'];
+    const cssColor = color.replace(' ', '').replace(color.charAt(0), color.charAt(0).toLowerCase());
+    let colorsList = [];
+    colorsList = mainPalette.map((mainValue) => getColorBlock(styles, cssColor, mainValue));
+
+    if (showAltPalette) {
+      altPalette.forEach((altValue) => {
+        colorsList.push(getColorBlock(styles, cssColor, altValue));
+      });
+    }
+
+    return (
+      <ul className={classes.colorGroup} key={cssColor}>
+        {getColorBlock(styles, cssColor, 500, color)}
+        <div className={classes.blockSpace} />
+        {colorsList}
+      </ul>
+    );
+  }
+
+  function getColorBlock(styles, colorName, colorValue, colorTitle) {
+    const bgColorText = colorName + colorValue;
+    const bgColor = colors[colorName][colorValue];
+    let fgColor = colors.fullBlack;
+    const contrastRatio = getContrastRatio(bgColor, fgColor);
```

Comment:
No need for an intermediary variable for that. It's creating an indirection with no logic/semantic value.